### PR TITLE
174 pf15 orwdx2 dcreason

### PR DESCRIPTION
**Pull request by kekuso**
- add tests and implementation for pf15 ORWDX2 DCREASON

### CODE REVIEW

**Comment by sidjtd** [comment]
jasmine_template has a console.log and lot of tests with xit in front of it... And line 75 with the spec file!  X'd out but tests left in.. by design?

Also you actually remembered to include all your files unlike my rpc from a second ago so.... props  :P

**Comment by kekuso** [comment]
oh the jasmine template is just like my personal flavor of boilerplate code for when I first start testing an rpc lol.
And I added the x's because of how my environment is set up. I might need to uncomment or comment out some lines of code from KD's abstractRpcEmulator files...

**Comment by sidjtd** [comment]
Oh cool, gotcha! I'll have to ask you questions on that later! Maybe after To chao (getting hungry and giving less pho'cs)

File: prototypes/fileparameters/pf15_orwdx2Dcreason.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        description: 'RPC to return a list of valid discontinuation reasons.',
+        returnType: 'ARRAY',
+        inputParameters: [],
+
+        processRPCInvocation(rpcArgs) {
+            let res = [];
+            res.push('~DCReason');
+            // get all ORDER REASON files
+            const ordReasonIenList = emulatorUtils.selectIENListFromFileType('100_03');
+            const ordReasonList = ordReasonIenList.filter((ien) => {
```
Comment (mfuroyama) [inline prototypes/fileparameters/pf15_orwdx2Dcreason.js]: This is great! Just an thought: you *might* want to consider adding a `map` step here on the `ordReasonList` array to convert IENs to VDM objects. before you do the `filter`. So your functional chain becomes: IENs -> map (to VDM objects) -> filter -> map.

This way, your final `map` step will be working with an array of VDM objects, and you don't have to do all those expensive `VDM.describes` again. It's a little bit more efficient that way.

Comment (kekuso) [inline prototypes/fileparameters/pf15_orwdx2Dcreason.js]: I love it

**Pull request merged**